beets/ui/__init__.py

```diff
@@ -99,6 +99,12 @@ def print_(*strings):
     replaces it.
     """
     if strings:
+        if not isinstance(strings[0], basestring):
+            try:
+                strings = map(unicode, strings)
+            except UnicodeError:
+                strings = map(bytes, strings)
```

Comment:
The goal was to allow statements such as `print_(my_album)`, which (1) is clean and comprehensible (2) matches the behaviour of the built-in `print` function, as you note.

It's true that `print_(object(), b'fo\xc3\xb6')` is ugly, but that will raise an `UnicodeError` → switch to bytes casting. Sure `print_(object(), b'fo\xc3\xb6', u'foö')` will still crash, but I think that's warranted. `print_(u'something', item)` will fail indeed, which matches the previous behaviour.

Possible solutions:

1. check each element of `strings` and convert the non-basestring ones. To bytes or unicode? Take the type of the other items (and raise an error if there's both bytes and unicodes in those other items).
2. remove the possibility to use multiple parameters. Pro: this makes the code simpler and surer. It's not a big deal there only one multiple-param `print_` invokation (`beets.ui.commands` l.1222 in `show_version()`). Con: that's not `print` behaviour.
3. delete `print_`: I feel like we should just always print unicode and let terminal emulators handle the rest. Such a brings surprisingly few problems here (just with info tests looking for paths in the output, fixed with 10 characters). However I've only got a linux box so I cannot vouch for the success on other platforms.